packages/hardhat-core/test/internal/core/providers/http.ts

```diff
@@ -1,7 +1,37 @@
+import { assert } from "chai";
+import { MockAgent, MockPool } from "undici";
+
 import { HttpProvider } from "../../../../src/internal/core/providers/http";
+import { SuccessfulJsonRpcResponse } from "../../../../src/internal/util/jsonrpc";
 import { ALCHEMY_URL } from "../../../setup";
 
 describe("HttpProvider", function () {
+  const url = "http://some.node";
+  const networkName = "NetworkName";
+  const mockPool = (() => {
```

Comment:
I find this pattern confusing tbh

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/2414/commits/5c50144dc2b807a25c6d9b4229145817c2b1022e